omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/VisitController1_9Test.java

```diff
@@ -214,4 +214,32 @@ public void searchByPatient_shouldGetUnretiredVisitsForThePatient() throws Excep
 		Assert.assertEquals(3, Util.getResultsSize(result));
 	}
 	
+	@Test
+	public void searchByPatient_shouldGetRetiredVisitsIfIncludeAllIsTrue() throws Exception {
+		String patientUUid = "da7f524f-27ce-4bb2-86d6-6d1d05312bd5";
+		int visitIncludingVoidedSize = 0;
+		int visitWithoutVoidedSize = 0;
+		
+		SimpleObject resultWithVoidedVisits = deserialize(handle(newGetRequest(getURI(),
+		    new Parameter("patient", patientUUid),
+		    new Parameter("includeAll", "true"))));
+		SimpleObject resultWithoutVoidedVisits = deserialize(handle(newGetRequest(getURI(),
+		    new Parameter("patient", patientUUid))));
+		
+		visitIncludingVoidedSize = Util.getResultsSize(resultWithVoidedVisits);
+		visitWithoutVoidedSize = Util.getResultsSize(resultWithoutVoidedVisits);
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("reason",
+		        "void test reason")));
+		
+		resultWithVoidedVisits = deserialize(handle(newGetRequest(getURI(),
+		    new Parameter("patient", patientUUid),
+		    new Parameter("includeAll", "true"))));
+		resultWithoutVoidedVisits = deserialize(handle(newGetRequest(getURI(),
+		    new Parameter("patient", patientUUid))));
+		
+		Assert.assertEquals(Util.getResultsSize(resultWithoutVoidedVisits), visitWithoutVoidedSize - 1);
```

Comment:
You know what? I think let us make this test easier to understand by using the actual expected count. For instance, if you expect 4 records, assert that they are so. If you expect 3 or 2 records, assert they are so.